### PR TITLE
Add Van-Component

### PR DESCRIPTION
**Pull request by Atmos4**
Simple function to create a standard Web Component from a VanJS component.

Example

```javascript
import van from "vanjs-core";
import { createComponent } from "vanjs-component";
const { button, div, slot } = van.tags;

createComponent("custom-counter", () => {
  const counter = van.state(0);
  return div(
    slot(),
    " ",
    counter,
    " ",
    button({ onclick: () => ++counter.val }, "+"),
    button({ onclick: () => --counter.val }, "-")
  );
});
```

and usage in HTML

```html
<custom-counter>❤️</custom-counter>

<custom-counter>👌</custom-counter>
```

### CODE REVIEW

File: addons/van_component/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+The web component will then hydrate properly with the counter, and slot the emoji correctly!
+
+## Why
+
+Hydration in VanJS is not convenient. This small function makes it trivial to have reusable van components. It also pairs very well with VanUI.
+
+## How to use
+
+This is very much WIP. The current way to use this is to copy the code in `dist`.
```
Comment (Tao-VanJS) [inline addons/van_component/README.md:51]: Are you able to publish an NPM for it for the ease of use?

File: addons/van_component/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+<custom-counter>👌</custom-counter>
+```
+
+## Create new elements from VanJS
+
+If you have another VanJS component, you can create web components with this syntax:
+
+```javascript
+const someComponent = (emoji) => van.tags["custom-counter"](emoji);
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply addons/van_component/README.md:40]: Yes that is correct!

File: addons/van_component/README.md (hunk 1 above)
Comment (Atmos4) [reply addons/van_component/README.md:51]: Done! The package is `vanjs-component`. I added a DTS file as well so DX should be decent

**Comment by Atmos4** [comment]
If that can help, I can work on an example page in the spirit of `Van Component by example`. Web components can be confusing so it might help others get a better understanding of what is possible!

**Comment by Tao-VanJS** [comment]
> If that can help, I can work on an example page in the spirit of `Van Component by example`. Web components can be confusing so it might help others get a better understanding of what is possible!

Yes, that would be great!

**Comment by Tao-VanJS** [comment]
@Atmos4,

I'm going to merging this PR first. If you want to add an example page in the future, you can send out a separate PR.

**Comment by Atmos4** [comment]
@Tao-VanJS I wanted to move at my own pace with this, so I took the liberty to create my own Git repo: https://github.com/Atmos4/van-element

Here are the changes I made since this PR:
- rename to `van-element`. The npm package is now `vanjs-element`.
- rename `createComponent` to `define`
- `define` type definition is now:
```ts
import { State, ChildDom } from "vanjs-core";

export declare const define: (
  name: string,
  element: (
    props: Record<string, State<string>> & {
      onMount: (mountFunction: (element: HTMLElement) => void) => void;
      element: HTMLElement;
    }
  ) => ChildDom,
  options?: { observed?: string[]; disableShadow?: boolean }
) => void;
```
- `element` method parameters:
  - a Record of attributes
  - `onMount` function (I plan to add `onDismount` or accept a return dismount function in future versions)
  - `element` which is the element itself
- `options`:
  - `observed` will make the component reactive to attribute changes.
  - `disableShadow` will disable Shadow DOM. This will turn the component into a normal custom element.
  
I will create a new PR to update this repo with the up-to-date code. 

Do you think it makes sense that I keep development on a separate repo, and merge code with this monorepo when I release a new version?

**Comment by Tao-VanJS** [comment]
> Do you think it makes sense that I keep development on a separate repo, and merge code with this monorepo when I release a new version?

Yeah, that looks good to me.

**Comment by Atmos4** [comment]
@Tao-VanJS here is the demo page: https://codepen.io/atmos4/pen/ZEPEvvB

I updated the [main repo](https://github.com/Atmos4/van-element). It is now in a state that I am happy about, so I will publish a PR soon! Don't hesitate to check it out and shout in issues if there are things you dislike

**Comment by Tao-VanJS** [comment]
Cool! This is great! Looking forward to your new PR.

**Pull request merged**